mtags/src/main/scala-2/scala/meta/internal/pc/PcCollector.scala

```diff
@@ -381,9 +379,9 @@ abstract class PcCollector[T](
          */
         case id: Ident
             if id.symbol == NoSymbol &&
-              soughtFilter(sought => sought.exists(_.name == id.name)) =>
+              soughtFilter(_.decodedName == id.name.decoded) =>
           fallbackSymbol(id.name, id.pos) match {
-            case Some(sym) if soughtFilter(sought => sought(sym)) =>
+            case Some(sym) if soughtFilter(_ == sym) =>
```

Comment:
In other places, we compare the symbols by `decodedName` but we do compare by `Symbol#==`, why we don't compare by decoded name here?

Reply:
I know the way of checking it is not very consistent, I will try to add a comment explaining it.
1. In most cases, we try to compare by `Symbol#==`.
2. The problem appears, where there is `NoSymbol` at a tree node. Then we check if the identifier there has the same decoded name. It it does, we look up the symbol at this position using `fallbackSymbol`, and again check if this time we got symbol equal by `Symbol#==`.

Reply:
Nnice,
What do you think about adding this comment https://github.com/scalameta/metals/pull/5100#discussion_r1157134931 to code comment or commit message?